arbor/backends/gpu/matrix_state_fine.hpp

```diff
@@ -475,14 +477,14 @@ struct matrix_state_fine {
     template <typename VFrom, typename VTo>
     void flat_to_packed(const VFrom& from, VTo& to ) {
         arb_assert(from.size()==matrix_size);
-        arb_assert(to.size()==data_size.back());
+        arb_assert(to.size()==data_size);
```

Comment:
I think that it could have been these calls here that were causing the original problem. They could have been called when a kernel was running on the GPU, accessing the same memory.